src/frontend/src/common/components/library/data_subview/components/DownloadModal/index.tsx

```diff
@@ -358,24 +360,43 @@ const DownloadModal = ({
 
     const onClick = () => {
       if (isFastaSelected) {
-        fastaDownloadMutation.mutate({
-          sampleIds: checkedSampleIds,
+        downloadMutation.mutate({
           endpoint: ORG_API.SAMPLES_FASTA_DOWNLOAD,
+          sampleIds: checkedSampleIds,
         });
       }
 
       if (isGisaidSelected) {
-        gisaidDownloadMutation.mutate({
+        downloadMutation.mutate({
+          endpoint: ORG_API.SAMPLES_FASTA_DOWNLOAD,
+          publicRepositoryName: PUBLIC_REPOSITORY_NAME.GISAID,
           sampleIds: checkedSampleIds,
-          endpoint: ORG_API.SAMPLES_GISAID_DOWNLOAD,
         });
+        const nPages = Math.ceil(nCompletedSampleIds / N_SAMPLES_PER_PAGE);
+        for (let page = 0; page < nPages; page++) {
+          downloadMutation.mutate({
+            endpoint: ORG_API.SAMPLES_TEMPLATE_DOWNLOAD,
+            page,
+            publicRepositoryName: PUBLIC_REPOSITORY_NAME.GISAID,
+            sampleIds: checkedSampleIds,
```

Comment:
checkedSamples needs to be subsetted to go along with the page, so for page0 we need checkedSamples[0-999]

Reply:
unless i'm missing something

Reply:
0-998, right? For 999 samples

Reply:
was it 999 or 1000? I'm having a hard time finding the supporting documentation for it. @danrlu can you confirm please?

Reply:
the limit is 1000, so per our last discussion ppl think we should play safe and just do 999 based on my memory, or I failed to communicate clearly XD Let me quickly double check and get back to you.

Reply:
thanks dan! let's keep it at 999 then just in case!

Reply:
This other ticket also uses 999: https://app.shortcut.com/genepi/story/194965/as-a-user-if-i-select-more-than-999-samples-i-want-to-be-notified-that-my-template-files-will-be-chopped-up